pymodbus/framer/rtu_framer.py

```diff
@@ -137,31 +129,38 @@ def isFrameReady(self):
 
         :returns: True if ready, False otherwise
         """
-        if len(self._buffer) > self._hsize:
-            if not self._header:
-                self.populateHeader()
+        if len(self._buffer) <= self._hsize:
+            return False
 
-            return self._header and len(self._buffer) >= self._header['len']
-        else:
+        try:
+            # Frame is ready only if populateHeader() successfully populates crc field which finishes RTU frame
+            # Otherwise, if buffer is not yet long enough, populateHeader() raises IndexError
+            self.populateHeader()
+        except IndexError:
```

Comment:
In case of incomplete frame `self.populateHeader()` raises `IndexError`. IndexError is used here as indicator that frame is not ready instead of using `len` key of `self._header` which can be not available in case of incomplete frame.